src/test/java/io/github/mfvanek/pg/model/ForeignKeyTest.java

```diff
@@ -46,4 +48,15 @@ void withInvalidArguments() {
         assertThrows(IllegalArgumentException.class, () -> ForeignKey.of("t", "c_t_order_id", null));
         assertThrows(IllegalArgumentException.class, () -> ForeignKey.of("t", "c_t_order_id", Collections.emptyList()));
     }
+
+    @Test
+    void testEquals() {
+        ForeignKey key1 = ForeignKey.of("t", "c_t_order_id", Arrays.asList("order_id", "limit"));
+        ForeignKey key2 = ForeignKey.of("t", "c_t_order_id", Arrays.asList("order_id", "limit"));
+        ForeignKey key3 = ForeignKey.of("t", "c_t_other_id", Arrays.asList("order_id", "limit"));
+
+        assertEquals(key1, key2);
+        assertNotEquals(key1, key3);
+        assertNotEquals(key2, key3);
```

Comment:
I think it would be better to also add checks for inequality of the columnsInConstraint field. Right now the test only checks equality/inequality of constraintName.